source/ast/symbols/MemberSymbols.cpp

```diff
@@ -2119,18 +2119,40 @@ std::span<const Expression* const> NetAliasSymbol::getNetReferences() const {
     auto syntax = getSyntax();
     SLANG_ASSERT(scope && syntax);
 
-    // TODO: there should be a global check somewhere that any given bit
-    // of a net isn't aliased to the same target signal bit multiple times.
     bitwidth_t bitWidth = 0;
     bool issuedError = false;
     SmallVector<const Expression*> buffer;
-    ASTContext context(*scope, LookupLocation::after(*this),
-                       ASTFlags::NonProcedural | ASTFlags::NotADriver);
+    ASTContext context(*scope, LookupLocation::after(*this), ASTFlags::NonProcedural);
+    auto& comp = context.getCompilation();
+    auto& aliasedSyms = comp.getNetAliases();
     NetAliasVisitor visitor(context);
 
-    for (auto exprSyntax : syntax->as<NetAliasSyntax>().nets) {
+    const auto nets = syntax->as<NetAliasSyntax>().nets;
+
+    // Preprocessing net alias expressions to relate them to existing net aliases.
+    const NetAliasSymbol* current = this;
+    for (auto exprSyntax : nets) {
```

Comment:
The logic here still seems wrong. Consider this simple example:
```
module m;
    wire [1:0] a, b, c;
    alias a = b[1:0];
    alias c = b[1:0];
endmodule
```
which results in the following error (when it should not):
```
source:5:15: error: it is not allowed to specify an alias from an individual signal to itself or to specify a given alias more than once
    alias c = b[1:0];
              ^~~~~~
source:4:15: note: alias declared here
    alias a = b[1:0];
              ^~~~~~
```

I don't think this mergeOrAddNetAlias idea works or is even necessary really. We don't need a compilation-wide map. Each net can already track which bits have been aliased using the driver mechanism. I think what you need to do is to do an addDriver() call across the cartesian product of the aliases in a given net alias statement. Instead of the "containing symbol" being the net alias, instead you make it the net it's being *aliased to*.

As an example:
```
alias a = b = c;
```
you want the following sequence of addDriver calls to be made, where the argument here is what gets set as the "containing symbol" inside ValueDriver:
a.addDriver(b);
a.addDriver(c);
b.addDriver(a);
b.addDriver(c);
c.addDriver(a);
c.addDriver(b);

And then I think the existing logic will just do the right thing without the need for a global map or comparing syntax nodes to each other.

Reply:
sorry for delay

solution was reimplemented according to your suggestion

Reply:
but i need to add more complex tests

Reply:
and fix CI

Reply:
CI was fixed and new tests were added

Reply:
CI was fixed and new tests were added

Reply:
CI was fixed and new tests were added